Rules/CommonScripts/PlayerCamera.as

```diff
@@ -205,3 +211,32 @@ void onRender(CRules@ this)
 		GUI::DrawText(text, ul, SColor(0xffffffff));
 	}
 }
+
+void onTick(CRules@ this)
+{
+	//initially position camera to view entire map
+	if (getGameTime() == 1)
+	{
+		CMap@ map = getMap();
+		CCamera@ camera = getCamera();
+		if (map !is null && camera !is null)
+		{
+			Vec2f mapDim = map.getMapDimensions();
+			pos = mapDim / 2;
+			camera.setPosition(pos);
+			camera.targetDistance = Maths::Min(calculateZoomLevelW(mapDim.x), calculateZoomLevelH(mapDim.y));
+		}
+	}
+
+	CControls@ controls = getControls();
+	if (controls !is null)
+	{
+		//enable cinematic camera
+		const int diffTime = deathTime - getGameTime();
+		if (controls.isKeyJustPressed(KEY_RBUTTON) && (spectatorTeam || diffTime <= 0))
+		{
+			SetTargetPlayer(null);
+			timeToCinematic = 0.0f;
```

Comment:
why are you automatically sending the player into cinematic mode everywhere in this file. At the very least there is a bug with the death lock which zooms in. When you set timeToCinematic = 0.0f; deathlock won't zoom in on the locked position because of the condition on line 128 of Spectator.as.

Reply:
ignore everything in this comment, rereading I was wrong about the bug. I do think however what you talked about was it not auto going to cinematic on some of those conditions but instead remembering.

Reply:
I don't understand what you mean. Death lock should still work like normal. Look at `PlayerCamera.as` from around line 135.